### PR TITLE
Change types scalable nodes

### PR DESCRIPTION
**Pull request by perezp**
In the current state there are modules with which there are autoscaling policies associated. However such modules do not belong to scalable types.

This PR addresses it, by changing the type of the modules that have associated autoscaling policies to seaclouds.nodes.ControlledDynamicWebAppCluster.

This PR also creates seaclouds.nodes.Compute type in node_types. 

Tests have been created to check that changes in the types have been produced in the correct nodes.

You can see that I've not squashed the commits. I've done it on purpose. I would like @rosogon looking at the two firsts because they are related to his PR that is currently failing. I thought it would be clearer for him if I do not squash all the changes into one already from the beginning.

### CODE REVIEW

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/OptimizerInitialDeployment.java [hunk 1]
```diff
@@ -124,11 +124,14 @@ public OptimizerInitialDeployment(SearchMethodName name) {
 
       Map<String, Object>[] appMapSolutions = hashMapOfFoundSolutionsWithThresholds(solutions, appMap, topology,
             appInfoSuitableOptions, numPlansToGenerate, requirements, suitableCloudOffer, hyst);
-
-      log.debug("Before ReplaceSuitableServiceByHost");
+      
+      
+      
+      log.debug("Before ReplaceSuitableServiceByHost and adding the seaclouds.nodes.Compute type information");
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/OptimizerInitialDeployment.java]: delete some line breaks

Comment (perezp) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/OptimizerInitialDeployment.java]: Done. 
Thanks.

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java [hunk 2]
```diff
@@ -483,9 +482,8 @@ private static Topology getApplicationTopologyRecursive(String elementName, Map<
       double hostPerformance = appInfoSuitableOptions.getCloudCharacteristics(elementName,
             YAMLmodulesOptimizerParser.getMeasuredPerformanceHost(elementName, groups)).getPerformance();
 
-      boolean elementCanScale= YAMLmodulesOptimizerParser.getScalabilityCapabilitiesOfModule(modules, elementName);
-            
-            
+      boolean elementCanScale = YAMLmodulesOptimizerParser.getScalabilityCapabilitiesOfModule(modules, elementName);
+
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java]: Below, I think this [if-else](https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/195/files#diff-66a9293b4c1da2179fae3c61935da42cR498) could be refactor to something like.
```
if (!topology.contains(moduleReqName)) {
  topology = getApplicationTopologyRecursive(moduleReqName, (Map<String, Object>)
              modules.get(moduleReqName), topology, modules, groups, appInfoSuitableOptions);
}

double opProfileBetweenModules = YAMLgroupsOptimizerParser.getOpProfileWithModule(elementName,
                   moduleReqName, groups);	 
newelement.addElementCalled(
                   new TopologyElementCalled(topology.getModule(moduleReqName), opProfileBetweenModules));           
```

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java [hunk 3]
```diff
@@ -714,6 +698,21 @@ public   static String getInitialElementName(Map<String, Object> appMap) {
       return groupsMap;
    }
 
+   public static Map<String, Object> getTypesMapFromAppMap(Map<String, Object> appMap) {
+      Map<String, Object> typesMap;
+      try {
+         if (log.isDebugEnabled()) {
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java]: I am not sure if it is useful `log.isDebugEnabled()` here. Take a look [here](http://stackoverflow.com/questions/963492/in-log4j-does-checking-isdebugenabled-before-logging-improve-performance)

File: planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+      try {
+         suitableCloudOffer = filenameToString(TestConstants.CLOUD_OFFER_FILENAME_IN_JSON);
+      } catch (IOException e) {
+         log.error("File for Cloud Offers not found");
+         e.printStackTrace();
+      }
+
+   }
+
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java]: I think this code is repeated by `OptimizerComputeTypeTest`. You could create a super class for avoiding duplicate code.

File: planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+   private static Optimizer optimizer;
+   private static String    appModel;
+   private static String    suitableCloudOffer;
+
+   static Logger log;
+
+   @BeforeClass
+   public void createObjects() {
+
+      log = LoggerFactory.getLogger(OptimizerTOSCADecember2015AutoscalingPoliciesTest.class);
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java]: `log` should be initialized when was defined [above](https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/195/files#diff-b3081f48199b3982c963add464a68e34R51).

File: planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+         e.printStackTrace();
+      }
+
+   }
+
+   private static String filenameToString(String path) throws IOException {
+      byte[] encoded = Files.readAllBytes(Paths.get(path));
+      return new String(encoded, StandardCharsets.UTF_8);
+   }
+
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java]: same above

File: planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+      } catch (IOException e) {
+         e.printStackTrace();
+      } finally {
+         if (out != null) {
+            out.close();
+         }
+      }
+
+   }
+
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java]: same above

File: planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+   @AfterClass
+   public void testFinishced() {
+      log.info("===== ALL TESTS FOR OPTIMIZER USING DECEMBER 2015 TOSCA FINISHED ===");
+   }
+
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java]: same above

**Comment by kiuby88** [comment]
Thx @perezp. Good job.
I have added some minor comments. could yo take a look?

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java (hunk 2 above)
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java]: Edited.
@perezp

Comment (perezp) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java]: I think so too. 
Done!
Thanks.

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java (hunk 3 above)
Comment (perezp) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java]: Removed the isDebugEnabled(). 
Thanks.

File: planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java (hunk 4 above)
Comment (perezp) [inline planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java]: Done. 
Thanks. 

This refactoring also addresses the next comments "same as above".

**Comment by kiuby88** [comment]
+1

**Comment by perezp** [comment]
Thanks for the review @kiuby88

**Pull request merged**